src/data/mystery-encounters/encounters/global-trade-system-encounter.ts

```diff
@@ -221,21 +224,48 @@ export const GlobalTradeSystemEncounter: MysteryEncounter =
             const tradePokemon = new EnemyPokemon(scene, randomTradeOption, pokemon.level, TrainerSlot.NONE, false);
             // Extra shiny roll at 1/128 odds (boosted by events and charms)
             if (!tradePokemon.shiny) {
-              // 512/65536 -> 1/128
-              tradePokemon.trySetShinySeed(512, true);
+              const baseShinyChance = 512;
+              const shinyThreshold = new Utils.IntegerHolder(baseShinyChance);
+              if (scene.eventManager.isEventActive()) {
+                shinyThreshold.value *= scene.eventManager.getShinyMultiplier();
+              }
+              scene.applyModifiers(ShinyRateBoosterModifier, true, shinyThreshold);
+
+              // Base shiny chance of 512/65536 -> 1/128, affected by events and Shiny Charms
+              // Maximum shiny chance of 4090/65536 -> 1/16, cannot improve further after that
+              const shinyChance = Math.min(shinyThreshold.value, 4090);
+
+              tradePokemon.trySetShinySeed(shinyChance, false);
             }
 
             // Extra HA roll at base 1/64 odds (boosted by events and charms)
-            if (pokemon.species.abilityHidden) {
-              const hiddenIndex = pokemon.species.ability2 ? 2 : 1;
-              if (pokemon.abilityIndex < hiddenIndex) {
+            const hiddenIndex = tradePokemon.species.ability2 ? 2 : 1;
+            if (tradePokemon.species.abilityHidden) {
+              if (tradePokemon.abilityIndex < hiddenIndex) {
                 const hiddenAbilityChance = new IntegerHolder(64);
                 scene.applyModifiers(HiddenAbilityRateBoosterModifier, true, hiddenAbilityChance);
 
                 const hasHiddenAbility = !randSeedInt(hiddenAbilityChance.value);
 
                 if (hasHiddenAbility) {
-                  pokemon.abilityIndex = hiddenIndex;
+                  tradePokemon.abilityIndex = hiddenIndex;
+                }
+              }
+            }
+
+            // If Pokemon is still not shiny or with HA, give the Pokemon a random Common egg move in its moveset
+            if (!tradePokemon.shiny && (!tradePokemon.species.abilityHidden || tradePokemon.abilityIndex < hiddenIndex)) {
+              const eggMoves: Moves[] = speciesEggMoves[tradePokemon.getSpeciesForm().getRootSpeciesId()];
```

Comment:
NIT: 
There is [Pokemon#getEggMoves()](https://github.com/pagefaultgames/pokerogue/blob/abb8abcceff3a566c0f4506cd278f8574f88917e/src/field/pokemon.ts#L1797-L1804)

```suggestion
              const eggMoves = tradePokemon.getEggMoves();
```